_posts/2024-02-05-cde-customization.adoc

```diff
@@ -0,0 +1,434 @@
+
+The following `ConfigMap` projects a Visual Studio Code `settings.json` file (remote VS Code data folder is `/checode/remote/data/machine`) that specifies a couple of Visual Studio Code Terminal settings:
+
+```bash
+kubectl apply -f - << EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vscode-settings
+  labels:
+    controller.devfile.io/mount-to-devworkspace: "true"
+    controller.devfile.io/watch-configmap: "true"
+  annotations:
+    controller.devfile.io/mount-as: file
+    controller.devfile.io/mount-path: /checode/remote/data/Machine/
+    controller.devfile.io/mount-access-mode: "511"
+data:
+  settings.json: |
+    {
+        "terminal.integrated.copyOnSelection": true,
+        "terminal.integrated.cursorBlinking": true
+    }
+EOF
+```
+
+The following `Secret` projects file `credentials` in fodler `/home/user/.aws`:
+
+```bash
+kubectl apply -f - << EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: aws-credentials
+  labels:
+    controller.devfile.io/mount-to-devworkspace: "true"
+    controller.devfile.io/watch-secret: "true"
+  annotations:
+    controller.devfile.io/mount-as: file
+    controller.devfile.io/mount-path: /home/user/.aws/
+    controller.devfile.io/mount-access-mode: "511"
+stringData:
+  credentials: |
+     [default]
+     aws_access_key_id     = ABCDEFGHILMNOPQRSTUVZ
+     aws_secret_access_key = ABCDEFGHILMNOPQRSTUVZ123456789+abcdefg
+EOF
+```
+
+### Override defaults configurations using a DevWorkspace Operator Configuration
+
+In this last section we want to show a last mechanism to override CDEs default properties. In this case using a https://doc.crds.dev/github.com/devfile/devworkspace-operator[DevWorkspaceOperatorConfig Custom Resource].
+
+`DevWorkspaceOperatorConfig` objects specifies advanced CDE properties such as the `Pod` `schedulerName`, if the `/home/user` folder will be persisted after a restart or not and many more.
+
+Applying a particular `DevWorkspaceOperatorConfig` to a CDEs takes 2 steps:
+1. Create the `DevWorkspaceOperatorConfig` custom resource in the Kubernetes cluster
+2. Edit the Devfile to specify to apply that specific configuration
+
+The following `DevWorkspaceOperatorConfig` override CDEs Pod `schedulerName` configuration:
+
+```bash
+kubectl apply -f - << EOF
+apiVersion: controller.devfile.io/v1alpha1
+kind: DevWorkspaceOperatorConfig
+metadata:
+  name: custom-dwoc
+  namespace: <namespace-name>
+config:
+  workspace:
+    schedulerName: my-scheduler
+EOF
+```
+
+To apply this configuration to a CDE it's required that the CDE Devfile references it:
+
+```diff
+schemaVersion: 2.2.0
+metadata:
+  name: rails-blog
++ attributes:
++   controller.devfile.io/devworkspace-config:
++     name: custom-dwoc
++     namespace: <namespace-name>
+components:
+...
+```
+
+With that attribute, the CDE started using this devfile, will use the scheduler `my-scheduler` (if it exist) rather then the default Kubernetes scheduler.
```

Comment:
```suggestion
With that attribute, the CDE started using this devfile will use the scheduler `my-scheduler` (if it exists) rather than the default Kubernetes scheduler.
```